### PR TITLE
Updated Ruby SDK documentation

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## What do these changes do?

> Ruby SDK doc updates

## Type of documentation

- [ ] New documentation
- [x] Documentation update
- [ ] Hotfix

### CODE REVIEW

File: docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx [hunk 1]
```diff
@@ -15,61 +17,48 @@ Check out the [**GitHub codebase**](https://github.com/rudderlabs/rudder-sdk-rub
...
 ## SDK setup requirements
 
-To set up the RudderStack Ruby SDK, the following prerequisites must be met:
-
-- You need to set up a [**RudderStack account**](https://app.rudderstack.com/signup).
-- Once signed up, <Link to="/dashboard-guides/sources/#adding-a-source">set up a Ruby source</Link> in the dashboard. You should be able to see a **Write Key** for this source, as shown below:
-
-<img src="../../../assets/event-stream-sources/ruby-write-key-new.png" alt="Ruby source write key" />
+1. Sign up to [RudderStack Cloud](https://app.rudderstack.com/signup).
+2. <Link to="/dashboard-guides/sources/#adding-a-source">Set up a Java source</Link> in your dashboard. You should be able to see a <strong>write key</strong> for this source.
```
Comment (ishachopra0401) [inline docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx:21]:
```suggestion
2. <Link to="/dashboard-guides/sources/#adding-a-source">Set up a Ruby source</Link> in your dashboard. You should be able to see a <strong>write key</strong> for this source.
```

File: docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx [hunk 2]
```diff
@@ -15,61 +17,48 @@ Check out the [**GitHub codebase**](https://github.com/rudderlabs/rudder-sdk-rub
...
 
-- You will also need a data plane URL. Refer to the <Link to="/resources/glossary/#data-plane-url">Glossary</Link> for more information on the data plane URL and where to find it.
+You will also need a data plane URL. Refer to the <Link to="/dashboard-guides/overview/#data-plane-url">Dashboard Overview</Link> guide for more information on the data plane URL and where to find it.
 
 <div class="successBlock">
-The <strong>Setup</strong> tab in the RudderStack dashboard (seen above) has an SDK installation snippet containing both the write key and the data plane URL. You can use it to integrate the Ruby SDK into your application.
+The <strong>Setup</strong> tab in the RudderStack dashboard (seen above) has the SDK installation snippet containing both the write key and the data plane URL. Copy it to integrate the Java SDK into your application.
```
Comment (ishachopra0401) [inline docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx:26]:
```suggestion
The <strong>Setup</strong> tab in the RudderStack dashboard (seen above) has the SDK installation snippet containing both the write key and the data plane URL. Copy it to integrate the Ruby SDK into your application.
```

File: docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx [hunk 3]
```diff
@@ -15,61 +17,48 @@ Check out the [**GitHub codebase**](https://github.com/rudderlabs/rudder-sdk-rub
...
 
 ```bash
 gem install rudder_analytics_sync
 ```
 
-## Using the SDK
+## Initializing the SDK
 
-To use the SDK, create a client instance as shown in the following snippet:
+To use the SDK, create a client instance as shown below:
```
Comment (ishachopra0401) [inline docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx:45]:
```suggestion
To initialize the SDK, create a client instance as shown below:
```

**Review by ishachopra0401 — APPROVED** [review]

File: docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx [hunk 4]
```diff
@@ -15,61 +17,48 @@ Check out the [**GitHub codebase**](https://github.com/rudderlabs/rudder-sdk-rub
...
 ```ruby
-require 'rudder_analytics_sync'
-
-Analytics = RudderAnalyticsSync::Client.new(
-  write_key: WRITE_KEY, # required
-  data_plane_url: DATA_PLANE_URL
+analytics = RudderAnalyticsSync::Client.new(
+  write_key: 'WRITE_KEY', # Required
+  data_plane_url: 'DATA_PLANE_URL',
+  stub: false,
```
Comment (pallabmaiti) [inline docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx:51]: Remove `stub`.

File: docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx [hunk 5]
```diff
@@ -82,71 +71,94 @@ Analytics.batch do |batch|
...
+
+The <Link to="/event-spec/standard-events/identify/">`identify`</Link> call lets you identify a visiting user and associate them to their actions. It also lets you record the traits about them like their name, email address, etc.
+
+A sample `identify` call made using the Ruby SDK is shown below:
 
 ```ruby
 Analytics.identify(
-  user_id: '12345',
+  user_id: '1hKOmRA4GRlm',
   traits: { email: "#{ username.email }", friends: 1 },
```
Comment (pallabmaiti) [inline docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx:113]: Add any random email.

File: docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx [hunk 6]
```diff
@@ -82,71 +71,94 @@ Analytics.batch do |batch|
...
-| **Field**      | **Type** | **Presence**                              | **Description**                                                                                                            |
-| :------------- | :------- | :---------------------------------------- | :------------------------------------------------------------------------------------------------------------------------- |
-| `anonymousId`  | String   | Optional                                  | Sets the user ID for cases where there is no unique identifier for the user. Either `userId` or `anonymousId` is required. |
-| `userId`       | String   | Optional, if `anonymousId` is already set | Unique identifier for a particular user in your database.                                                                  |
-| `context`      | Object   | Optional                                  | Dictionary of information that provides context about a message. However, it is not directly related to the API call.      |
-| `integrations` | Object   | Optional                                  | A dictionary containing the destinations to be either enabled or disabled.                                                 |
```
Comment (pallabmaiti) [inline docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx:111]: Add `integrations`.

File: docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx [hunk 7]
```diff
@@ -82,71 +71,94 @@ Analytics.batch do |batch|
...
-| `user_id`      | String | Required | The developer identification for your user                                                                                 |
-| `event`        | String | Required | Name of the event being performed by the user                                                                              |
-| `properties`   | Object | Optional | Dictionary of the properties associated with a particular event.                                                           |
-| `context`      | Object | Optional | Dictionary of information that provides context about a message. However, it is not directly related to the API call.      |
-| `timestamp`    | Date   | Optional | The timestamp of the message's arrival.                                                                                    |
-| `anonymous_id` | String | Optional | Sets the user ID for cases where there is no unique identifier for the user. Either `userId` or `anonymousId` is required. |
-| `integrations` | Object | Optional | A dictionary containing the destinations to be either enabled or disabled.                                                 |
```
Comment (pallabmaiti) [inline docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx:139]: Add `integrations`.

File: docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx [hunk 8]
```diff
@@ -155,71 +167,93 @@ Analytics.page(
...
-| **Field**      | **Type** | **Presence**                              | **Description**                                                                                                            |
-| :------------- | :------- | :---------------------------------------- | :------------------------------------------------------------------------------------------------------------------------- |
-| `anonymousId`  | String   | Optional                                  | Sets the user ID for cases where there is no unique identifier for the user. Either `userId` or `anonymousId` is required. |
-| `userId`       | String   | Optional, if `anonymousId` is already set | Unique identifier for a particular user in your database.                                                                  |
-| `context`      | Object   | Optional                                  | Dictionary of information that provides context about a message. However, it is not directly related to the API call.      |
-| `integrations` | Object   | Optional                                  | A dictionary containing the destinations to be either enabled or disabled.                                                 |
```
Comment (pallabmaiti) [inline docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx:163]: Add `integrations`.

File: docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx [hunk 9]
```diff
@@ -155,71 +167,93 @@ Analytics.page(
...
```
Comment (pallabmaiti) [inline docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx:189]: Add `integrations`.

File: docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx [hunk 10]
```diff
@@ -155,71 +167,93 @@ Analytics.page(
...
+Analytics.alias(previous_id: '1hKOmRA4GRlm', user_id: '12345')
 ```
 
 The `alias` method parameters are as mentioned below:
 
-| **Field**      | **Type** | **Presence**                              | **Description**                                                                                                       |
-| :------------- | :------- | :---------------------------------------- | :-------------------------------------------------------------------------------------------------------------------- |
-| `userId`       | String   | Optional, if `anonymousId` is already set | Unique identifier for a particular user in your database.                                                             |
-| `context`      | Object   | Optional                                  | Dictionary of information that provides context about a message. However, it is not directly related to the API call. |
-| `integrations` | Object   | Optional                                  | A dictionary containing the destinations to be either enabled or disabled.                                            |
```
Comment (pallabmaiti) [inline docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx:215]: Add `integrations`.

**Review by pallabmaiti — CHANGES_REQUESTED** [review]

File: docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx (hunk 4 above)
Comment (ameypv-rudder) [inline docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx:51]:
```suggestion
```

File: docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx (hunk 5 above)
Comment (ameypv-rudder) [inline docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx:113]:
```suggestion
  traits: { email: "alex@example.com", friends: 1 },
```

File: docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx [hunk 11]
```diff
@@ -82,71 +71,94 @@ Analytics.batch do |batch|
...
-| `traits`       | Object   | Optional                                  | Dictionary of the traits associated with the user, such as `name`or `email`.                                               |
+| **Field**      | **Type** |  **Description**     |
+| :------------- | :------- |  :---------------- |
+| `user_id` <br/> <span style="color: #4D4DFF;font-size:12px;">Required, if <strong>anonymous_id</strong> is absent.</span>     | String   | Unique identifier for a user in your database.      |
+| `anonymous_id`  <br/> <span style="color: #4D4DFF;font-size:12px;">Required, if <strong>user_id</strong> is absent.</span> | String   |  The SDK automatically sets this identifier in cases where there is no unique identifier for the user. |
+| `traits`       | Object   | An optional dictionary of the user's traits like `name` or `email`.     |
+| `context`      | Object   |  An optional dictionary of information that provides context about the event. It is not directly related to the API call.      |
```
Comment (ameypv-rudder) [inline docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx:125]:
```suggestion
| `context`      | Object   |  An optional dictionary of information that provides context about the event. It is not directly related to the API call.      |
| `integrations` | Object | Optional | An optional dictionary containing the destinations to be either enabled or disabled. |
```

File: docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx [hunk 12]
```diff
@@ -82,71 +71,94 @@ Analytics.batch do |batch|
...
-| `integrations` | Object | Optional | A dictionary containing the destinations to be either enabled or disabled.                                                 |
+| **Field**      | **Type** |  **Description**     |
+| :------------- | :------- |  :---------------- |
+| `user_id` <br/> <span style="color: #4D4DFF;font-size:12px;">Required, if <strong>anonymous_id</strong> is absent.</span>     | String   | Unique identifier for a user in your database.      |
+| `anonymous_id`  <br/> <span style="color: #4D4DFF;font-size:12px;">Required, if <strong>user_id</strong> is absent.</span> | String   |  The SDK automatically sets this identifier in cases where there is no unique identifier for the user. |
+| `event`  <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span>      | String | Name of the event.           |
+| `properties`   | Object | An optional dictionary of the properties associated with the event.   |
+| `context`      | Object | An optional dictionary of information that provides context about the event. It is not directly related to the API call.      |
+| `timestamp`    | Timestamp in ISO 8601 format     | The timestamp of the event's arrival.  |
```
Comment (ameypv-rudder) [inline docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx:151]:
```suggestion
| `timestamp`    | Timestamp in ISO 8601 format     | The timestamp of the event's arrival.  |
| `integrations` | Object | Optional | An optional dictionary containing the destinations to be either enabled or disabled. |
```

File: docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx [hunk 13]
```diff
@@ -155,71 +167,93 @@ Analytics.page(
...
-| `timestamp`    | Date     | Optional                                  | The timestamp of the message's arrival.                                                                                    |
+| **Field**      | **Type** |  **Description**     |
+| :------------- | :------- |  :---------------- |
+| `user_id` <br/> <span style="color: #4D4DFF;font-size:12px;">Required, if <strong>anonymous_id</strong> is absent.</span>     | String   | Unique identifier for a user in your database.      |
+| `anonymous_id`  <br/> <span style="color: #4D4DFF;font-size:12px;">Required, if <strong>user_id</strong> is absent.</span> | String   |  The SDK automatically sets this identifier in cases where there is no unique identifier for the user. |
+| `name`  <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span>      | String | Name of the viewed page.           |
+| `properties`   | Object | An optional dictionary of the properties associated with the viewed page, like `url` or `referrer`.   |
```
Comment (ameypv-rudder) [inline docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx:175]:
```suggestion
| `properties`   | Object | An optional dictionary of the properties associated with the viewed page, like `url` or `referrer`.   |
| `integrations` | Object | An optional dictionary containing the destinations to be either enabled or disabled. |
```

File: docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx [hunk 14]
```diff
@@ -155,71 +167,93 @@ Analytics.page(
...
-| `timestamp`    | Date     | Optional                                  | The timestamp of the message's arrival.                                                                                    |
-
-## Alias
+| **Field**      | **Type** |  **Description**     |
+| :------------- | :------- |  :---------------- |
+| `user_id` <br/> <span style="color: #4D4DFF;font-size:12px;">Required, if <strong>anonymous_id</strong> is absent.</span>     | String   | Unique identifier for a user in your database.      |
+| `anonymous_id`  <br/> <span style="color: #4D4DFF;font-size:12px;">Required, if <strong>user_id</strong> is absent.</span> | String   |  The SDK automatically sets this identifier in cases where there is no unique identifier for the user. |
+| `group_id`  <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span>    | String   | Unique identifier of the group in your database.    |
+| `traits`       | Object   | An optional dictionary of the group's traits like `name`or `email`.     |
```
Comment (ameypv-rudder) [inline docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx:230]:
```suggestion
| `traits`       | Object   | An optional dictionary of the group's traits like `name`or `email`.     |
| `integrations` | Object | An optional dictionary containing the destinations to be either enabled or disabled. |
```

File: docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx [hunk 15]
```diff
@@ -155,71 +167,93 @@ Analytics.page(
...
-
-For a detailed explanation of the <code class="inline-code">alias</code> call, refer to our <a href="https://rudderstack.com/docs/event-spec/standard-events/alias/">RudderStack API Specification</a> guide.
-</div>
-
+| **Field**      | **Type** |  **Description**     |
+| :------------- | :------- |  :---------------- |
+| `user_id` <br/> <span style="color: #4D4DFF;font-size:12px;">Required, if <strong>anonymous_id</strong> is absent.</span>     | String   | Unique identifier for a user in your database.      |
+| `anonymous_id`  <br/> <span style="color: #4D4DFF;font-size:12px;">Required, if <strong>user_id</strong> is absent.</span> | String   |  The SDK automatically sets this identifier in cases where there is no unique identifier for the user. |
+| `previous_id` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span>  | String   | The previous unique identifier of the user.        |
+| `traits`       | Object   | An optional dictionary of the user's traits like `name` or `email`.     |
```
Comment (ameypv-rudder) [inline docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx:255]:
```suggestion
| `traits`       | Object   | An optional dictionary of the user's traits like `name` or `email`.     |
| `integrations` | Object | An optional dictionary containing the destinations to be either enabled or disabled. |
```

File: docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx [hunk 16]
```diff
@@ -155,71 +167,93 @@ Analytics.page(
...
+
+The `screen` method parameters are as described below:
 
-## Group
+| **Field**      | **Type** |  **Description**     |
+| :------------- | :------- |  :---------------- |
+| `user_id` <br/> <span style="color: #4D4DFF;font-size:12px;">Required, if <strong>anonymous_id</strong> is absent.</span>     | String   | Unique identifier for a user in your database.      |
+| `anonymous_id`  <br/> <span style="color: #4D4DFF;font-size:12px;">Required, if <strong>user_id</strong> is absent.</span> | String   |  The SDK automatically sets this identifier in cases where there is no unique identifier for the user. |
+| `name`  <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span>      | String | Name of the viewed screen.           |
+| `properties`   | Object | An optional dictionary of the properties associated with the viewed screen, like `url` or `referrer`.   |
```
Comment (ameypv-rudder) [inline docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx:205]:
```suggestion
| `properties`   | Object | An optional dictionary of the properties associated with the viewed screen, like `url` or `referrer`.   |
| `integrations` | Object | An optional dictionary containing the destinations to be either enabled or disabled. |
```

File: docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx (hunk 4 above)
Comment (ameypv-rudder) [reply docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx]: Done.

File: docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx [hunk 17]
```diff
@@ -82,71 +70,96 @@ Analytics.batch do |batch|
...
+The Gzip feature is enabled by default in the Ruby SDK version 2.0.0.
+</div>
+
+The Ruby SDK automatically gzips requests. However, you can disable this feature by setting the `Gzip` parameter to `false` while initializing the SDK, as shown:
+
+```ruby
+analytics = RudderAnalyticsSync::Client.new(
+  write_key: 'WRITE_KEY', # required
+  data_plane_url: 'DATA_PLANE_URL',
+  stub: false,
```
Comment (pallabmaiti) [inline docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx:85]: Remove `stub`.

**Review by pallabmaiti — CHANGES_REQUESTED** [review]

File: docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx (hunk 17 above)
Comment (ameypv-rudder) [inline docs/sources/event-streams/sdks/rudderstack-ruby-sdk.mdx:85]:
```suggestion
```

**Review by pallabmaiti — APPROVED** [review]

**Pull request merged**